include/arch/arm/arch/64/mode/object/structures.h

```diff
@@ -26,8 +26,7 @@ typedef struct arch_tcb {
 enum vm_rights {
     VMKernelOnly = 0,
     VMReadWrite = 1,
-    VMKernelReadOnly = 2,
-    VMReadOnly = 3
+    VMReadOnly = 2,
```

Comment:
I don't think we should change the value of `VMReadOnly` -- these can be used as bit patterns, i.e. `vm_rights_t` below is just `word` and is probably used in PTEs etc.

Which might explain why there is a value here at all for 2, it might just be a pattern we don't use, but where it would look slightly strange to have a jump in the enum. I'd still be happy with that jump and removing the name, because it does suggest to the user that it does something, which is not the case.

Reply:
I thought that the value of a `vm_rights_t` was not directly used anywhere, for example for the PTE encoding `APFromVMRights` is used.

Reply:
You're right, I didn't trace that to the very end, we do appear to only use it symbolically. As long as we don't have C verification I'm still reluctant -- if we've overlooked anything this will break stuff, whereas if we keep the value we can be certain that we are not changing behaviour.

Also from the user perspective, changing this value breaks the ABI. Not that we usually care that much about this aspect, but if it's easy, why not keep it compatible. We have a lot of breaking changes already in this release, better not to introduce more subtle things that might go wrong for people.

Reply:
I don't see how it breaks the ABI as I think this is an enum internal to the kernel and cannot be used by user-space.

Regardless, I understand the value in not changing the value and will drop that change.

Reply:
Oh, you are totally right, I should really not stop tracing things after each point. The API and ABI is via `seL4_CapRights_t`, so this is not a breaking change either way.

So the only concern that would remain with changing the value is being sure that we are right that it's used as pure enum only.